### PR TITLE
Toscalib

### PR DESCRIPTION
**Pull request by mchasal**
Interim code checkpoint.

### CODE REVIEW

File: juju-tosca-import.py [hunk 1]
```diff
@@ -44,27 +45,15 @@ def description():
...
-    yc = yaml.load(content, Loader=Loader)
-    return yc
-
-
 def unpack_zip(zipfn):
     # zip file needs to be in the TOSCA CSAR format
+    # TODO This may be unnecessary if/when toscalib does it
     try:
         zip = zipfile.ZipFile(zipfn, 'r')
     except:
```
Comment (wwitzel3) [inline juju-tosca-import.py]: catch the actual exception, this is esspecially important for command-line interactions. Using a bar exception can prevent keyboard interupts from functioning properly as well as mask underlying problems with an implementation.

File: juju-tosca-import.py [hunk 2]
```diff
@@ -73,47 +62,174 @@ def unpack_zip(zipfn):
...
 def parse_metafile(tmpdir):
+    # Parse the TOSCA.meta file looking for yaml definitions
+    # TODO This may be unnecessary if/when toscalib does it
     if not os.path.isfile(tmpdir+"/TOSCA-Metadata/TOSCA.meta"):
-        print "TOSCA.meta not found in CSAR file"
+        print "Error: TOSCA.meta not found in CSAR file"
+        sys.exit(1)
+    try:
+        tfile = open(tmpdir+'/TOSCA-Metadata/TOSCA.meta', 'r')
+    except:
```
Comment (wwitzel3) [inline juju-tosca-import.py]: catch the actual exception

File: juju-tosca-import.py [hunk 3]
```diff
@@ -73,47 +62,174 @@ def unpack_zip(zipfn):
...
+    # Make dirs and open files
+    charmdir = bundledir + "/charms/tosca." + nodetmp.name + "-1"
+    if not os.path.exists(charmdir):
+        os.makedirs(charmdir)
+    if not os.path.exists(charmdir + "/hooks"):
+        os.makedirs(charmdir + "/hooks")
+    configfn = charmdir + "/config.yaml"
+    try:
+        configfile = open(configfn, 'w')
+    except:
```
Comment (wwitzel3) [inline juju-tosca-import.py]: catch the actual exception

File: juju-tosca-import.py [hunk 4]
```diff
@@ -73,47 +62,174 @@ def unpack_zip(zipfn):
...
+        print ("Error: Couldn't open config.yaml")
+        sys.exit(1)
+    metafn = charmdir + "/metadata.yaml"
+    try:
+        metafile = open(metafn, 'w')
+    except:
```
Comment (wwitzel3) [inline juju-tosca-import.py]: catch the actual exception

File: juju-tosca-import.py [hunk 5]
```diff
@@ -73,47 +62,174 @@ def unpack_zip(zipfn):
...
+        print ("Error: Couldn't open metadata.yaml")
+        sys.exit(1)
+    myaml = {}
```
Comment (wwitzel3) [inline juju-tosca-import.py]: myaml = {
    'name': nodetmp.name,
    'summary': 'juju-tosca imported charm',
    'description': 'juju-tosca imported charm',
  }

File: juju-tosca-import.py [hunk 6]
```diff
@@ -73,47 +62,174 @@ def unpack_zip(zipfn):
...
+
+    metafile.write(
+        yaml.safe_dump(myaml, default_flow_style=False, allow_unicode=True)
+    )
+    metafile.close()
+
+    # Create the hooks
+    cyaml = {}
+    for int in nodetmp.interfaces:
+        if int.type == "tosca.interfaces.node.Lifecycle":
```
Comment (wwitzel3) [inline juju-tosca-import.py]: Lets consider a refactor here.
- http://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html

File: juju-tosca-import.py (hunk 1 above)
Comment (mchasal) [inline juju-tosca-import.py]: Added specific exceptions.

File: juju-tosca-import.py (hunk 2 above)
Comment (mchasal) [inline juju-tosca-import.py]: Added specific exceptions.

File: juju-tosca-import.py (hunk 3 above)
Comment (mchasal) [inline juju-tosca-import.py]: Added specific exceptions.

File: juju-tosca-import.py (hunk 4 above)
Comment (mchasal) [inline juju-tosca-import.py]: Added specific exceptions.

File: juju-tosca-import.py (hunk 5 above)
Comment (mchasal) [inline juju-tosca-import.py]: Updated to the suggested style of building the dict. Also found a couple of other places and made them consistent.

File: juju-tosca-import.py (hunk 6 above)
Comment (mchasal) [inline juju-tosca-import.py]: Refactored to match that suggestion, though I'm not 100% it's as you were asking, so please let me know if it's not.

**Pull request merged**